lib/stylelint/plugin.rb

```diff
@@ -1,34 +1,117 @@
-  # the public interface documented. Danger uses [YARD](http://yardoc.org/)
-  # for generating documentation from your plugin source, and you can verify
-  # by running `danger plugins lint` or `bundle exec rake spec`.
+  # Lint javascript files using [stylelint](http://stylelint.io/).
+  # Results are send as inline comment.
   #
-  # You should replace these comments with a public description of your library.
+  # @example Run stylelint with changed files only
   #
-  # @example Ensure people are well warned about merging on Mondays
+  #          stylelint.filtering = true
+  #          stylelint.lint
   #
-  #          my_plugin.warn_on_mondays
-  #
-  # @see  Rossukhon Leagmongkol/danger-stylelint
-  # @tags monday, weekends, time, rattata
+  # @see  nimblehq/danger-stylelint
+  # @tags lint, stylelint, css, scss, sass
   #
   class DangerStylelint < Plugin
-    # An attribute that you can read/write from your Dangerfile
+    DEFAULT_BIN_PATH = "./node_modules/.bin/stylelint"
+
+    # An path to stylelint's config file
+    # @return [String]
+    attr_accessor :config_file
+
+    # Enable filtering
+    # Only show messages within changed files.
+    # @return [Boolean]
+    attr_accessor :filtering
+
+    # A path of stylelint's bin
+    # @return [String]
+    attr_writer :bin_path
+
+    def bin_path
+      @bin_path ||= DEFAULT_BIN_PATH
+    end
+
+    # Specify extensions of target file
+    # Default is [".css, .scss, .sass"]
+    # @return [Array]
+    attr_writer :target_extensions
+
+    def target_extensions
+      @target_extensions ||= %w(.css .scss .sass)
+    end
+
+    # Lints css files.
+    # Generates `errors` and `warnings` according to stylelint's config.
     #
-    # @return   [Array<String>]
-    attr_accessor :my_attribute
+    # @return  [void]
+    #
+    def lint
+      lint_results
+        .reject { |result| result.nil? || result["warnings"].length.zero? }
+        .map { |result| send_comment result }
+    end
+
+    private
+
+    # Get stylelint bin path
+    #
+    # return [String]
+    def stylelint_path
+      File.exist?(bin_path) ? bin_path : find_executable("stylelint")
+    end
+
+    # Get all lintable files
+    #
+    # return [String]
+    def all_lintable_files
+      "**/*{#{target_extensions.join(',')}}"
+    end
+
+    # Get lint result regards the filtering option
+    #
+    # return [Hash]
+    def lint_results
+      bin = stylelint_path
+      raise "stylelint is not installed" unless bin
+
+      return run_lint(bin, all_lintable_files) unless filtering
+
+      ((git.modified_files - git.deleted_files - git.renamed_files.map { |r| r[:before] }) + git.added_files + git.renamed_files.map { |r| r[:after] })
+        .select { |f| target_extensions.include?(File.extname(f)) }
+        .map { |f| f.gsub("#{Dir.pwd}/", "") }
+        .map { |f| run_lint(bin, f).first }
+    end
+
+    # Run stylelint against a single file.
```

Comment:
How about

```suggestion
    # Run stylelint against file(s).
```
as It could be run against multiple files as well from this part

```ruby
return run_lint(bin, all_lintable_files) unless filtering
```

Reply:
done in 4f636e6f6e7e07faf138ee91211bdca380637e61!